eth/hooks/engine_v2_hooks.go

```diff
@@ -28,8 +29,25 @@ func AttachConsensusV2Hooks(adaptor *XDPoS.XDPoS, bc *core.BlockChain, chainConf
 		listBlockHash[0] = currentHash
 		parentNumber := number.Uint64() - 1
 		parentHash := currentHash
+
+		// check and wait the latest block is already in the disk
+		// sometimes blocks are yet inserted into block
+		for timeout := 0; ; timeout++ {
+			parentHeader := chain.GetHeaderByNumber(parentNumber)
```

Comment:
why don't use parentHash and parentNumber together?

Reply:
👍